File: apps/ios/self-management/Features/Diary/Domain/Entities/DiaryMood.swift
//
//  DiaryMood.swift
//  self-management
//

import Foundation

/// Represents the emotional state associated with a diary entry.
public enum DiaryMood: String, CaseIterable, Codable {
    case happy      = "happy"
    case productive = "productive"
    case tired      = "tired"
    case neutral    = "neutral"
    
    /// Display name for the mood
    public var displayName: String {
        self.rawValue.capitalized
    }
    
    /// Emoji associated with the mood
    public var emoji: String {
        switch self {
        case .happy: return "😊"
        case .productive: return "🚀"
        case .tired: return "😴"
        case .neutral: return "😐"
        }
    }
}

File: apps/mobile/lib/features/diary/domain/entities/diary_mood.dart
enum DiaryMood {
  happy,
  productive,
  tired,
  neutral;

  String get displayName {
    switch (this) {
      case DiaryMood.happy:
        return 'Happy';
      case DiaryMood.productive:
        return 'Productive';
      case DiaryMood.tired:
        return 'Tired';
      case DiaryMood.neutral:
        return 'Neutral';
    }
  }

  String get emoji {
    switch (this) {
      case DiaryMood.happy:
        return '😊';
      case DiaryMood.productive:
        return '🚀';
      case DiaryMood.tired:
        return '😴';
      case DiaryMood.neutral:
        return '😐';
    }
  }
}
